python/triton/compiler/compiler.py

```diff
@@ -415,8 +415,10 @@ def _init_handles(self):
             if self.metadata.tmem_size > max_tmem_size:
                 raise OutOfResources(self.metadata.tmem_size, max_tmem_size, "tensor memory")
         # TODO: n_regs, n_spills should be metadata generated when calling `ptxas`
-        self.module, self.function, self.n_regs, self.n_spills = driver.active.utils.load_binary(
+        self.module, self.function, self.n_regs, self.n_spills, self.n_threads = driver.active.utils.load_binary(
             self.name, self.kernel, self.metadata.shared, device)
+        if self.metadata.num_warps * 32 > self.n_threads:
```

Comment:
`* 32` can be wrong for AMD GPUs

Reply:
+1. You can see https://github.com/triton-lang/triton/blob/65f646c52a906a57311f260f92d18fe960e3715c/third_party/amd/backend/compiler.py#L72-L81 for details.